sys/usb/usbus/hid/hid_io.c

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright (C) 2020 Nils Ollrogge
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup usbus_hid
+ * @{
+ * @file
+ * @brief This file implements a USB HID callback and read/write functions.
+ *
+ * @author  Nils Ollrogge <nils-ollrogge@outlook.de>
+ * @}
+ */
+
+#define USB_H_USER_IS_RIOT_INTERNAL
+
+#include <string.h>
+
+#include "isrpipe.h"
+#include "isrpipe/read_timeout.h"
+
+#include "usb/usbus.h"
+#include "usb/usbus/hid.h"
+
+#define ENABLE_DEBUG    (0)
+#include "debug.h"
+
+static usbus_hid_device_t hid;
+static uint8_t _hid_rx_buf_mem[CONFIG_USBUS_HID_INTERRUPT_EP_SIZE];
+static isrpipe_t _hid_stdio_isrpipe = ISRPIPE_INIT(_hid_rx_buf_mem);
+
+int usb_hid_io_read(void *buffer, size_t size)
+{
+    return isrpipe_read(&_hid_stdio_isrpipe, buffer, size);
+}
+
+int usb_hid_io_read_timeout(void *buffer, size_t size, uint32_t timeout)
+{
+    return isrpipe_read_timeout(&_hid_stdio_isrpipe, buffer, size, timeout);
+}
+
+void usb_hid_io_write(const void *buffer, size_t len)
+{
+    uint8_t* buffer_ep = hid.iface.ep->next->ep->buf;
```

Comment:
same comment here.
It would be better to avoid ep->next->ep but rather directly point to the right endpoint.
You could add a pointer to the rightful endpoint in your usbus_hid_device_t struct.

Reply:
I don't know if this is a good idea, because the HID endpoint can have a variable amount of endpoints.

Reply:
I understand what you mean but on the other side, if we add or modify the endpoints registration during the hid initialization, we might break this.